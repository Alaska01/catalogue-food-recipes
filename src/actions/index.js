import axios from 'axios';

const GET_RECIPES = () => (dispatch) => {
  const allCategories = ['Beef', 'Chicken', 'Dessert', 'Vegan', 'Vegetarian'];

  const allCategoriesResult = [];

  // eslint-disable-next-line array-callback-return
  allCategories.map((category) => {
    axios.get(`https://www.themealdb.com/api/json/v1/1/filter.php?c=${category}`)
      .then((res) => {
        allCategoriesResult.push(...res.data.meals);
      }).then(() => {
        dispatch({
          type: 'GET RECIPES',
          payload: allCategoriesResult,
        });
      });
  });
};

const GET_RECIPE = (recipeId) => async (dispatch) => {
  const data = await axios.get(`https://www.themealdb.com/api/json/v1/1/lookup.php?i=${recipeId}`);
  dispatch({
    type: 'GET RECIPE',
    payload: data.data.meals[0],
  });
};

const FILTER_RECIPELIST = (category) => async (dispatch) => {
  const data = await axios.get(`https://www.themealdb.com/api/json/v1/1/filter.php?c=${category}`);
  dispatch({
    type: 'FILTER CATEGORY',
    payload: data,
  });
};

const CLEAR_DATA = () => (dispatch) => {
  dispatch({
    type: 'RESET_DATA',
  });
};

export {
  GET_RECIPES, FILTER_RECIPELIST, GET_RECIPE, CLEAR_DATA,
};
