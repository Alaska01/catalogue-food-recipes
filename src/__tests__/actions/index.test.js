import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { GET_RECIPE, CLEAR_DATA } from '../../actions/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('async actions', () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
  });

  it('dispatches correct action and payload', () => {
    store.dispatch(GET_RECIPE());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches correct action and payload', () => {
    store.dispatch(CLEAR_DATA());
    expect(store.getActions()).toMatchSnapshot();
  });
});
