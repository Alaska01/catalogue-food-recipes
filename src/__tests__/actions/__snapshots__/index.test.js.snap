// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions dispatches correct action and payload 1`] = `Array []`;

exports[`async actions dispatches correct action and payload 2`] = `
Array [
  Object {
    "type": "RESET_DATA",
  },
]
`;
